application/tests/cheatsheets_workstream_f_test.py

```
@@ -122,5 +126,148 @@ def test_written_file_validates_against_schema(self) -> None:
+
+    def test_unknown_cre_id_skipped_sibling_still_linked(self) -> None:
+        cache = _StubCache({"764-507": _cre("764-507")})
+        # One known sibling, one unknown id on the same suggestion.
+        approved = [_sugg("Authentication Cheat Sheet", ["764-507", "999-999"])]
+
+        with self.assertLogs(wf.logger, level="WARNING") as cm:
+            result = wf.suggestions_to_parse_result(approved, cache)
+
+        standards = result.results["OWASP Cheat Sheets"]
+        self.assertEqual(len(standards), 1)
+        self.assertEqual([l.document.id for l in standards[0].links], ["764-507"])
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)